homeassistant/components/efergy/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow for Efergy integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pyefergy import Efergy, exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_APPTOKEN, DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EfergyFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Efergy."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            api_key = user_input[CONF_API_KEY]
+
+            self._async_abort_entries_match({CONF_API_KEY: api_key})
+            hid, error = await self._async_try_connect(api_key)
+            if error is None:
+                entry = await self.async_set_unique_id(hid)
+                if entry:
+                    self.hass.config_entries.async_update_entry(entry, data=user_input)
+                    await self.hass.config_entries.async_reload(entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=DEFAULT_NAME,
+                    data={CONF_API_KEY: api_key},
+                )
+            errors["base"] = error
+
+        user_input = user_input or {}
```

Comment:
Should we use this to set defaults in the form? Otherwise there's no point to this line I think.